src/main/java/com/starbucks/service/OrderService.java

```diff
@@ -18,39 +17,31 @@ public OrderService(OrderRepository orderRepository) {
     }
 
     public String saveOrder(Order order) {
-
-        orderRepository.save ( order );
-        return order.toString ( );
+        orderRepository.save(order);
+        return order.toString();
     }
 
-    public String updateOrder(Order order , UUID id) {
-
-        if (orderRepository.existsById ( id )) {
-            Order orderDB = orderRepository.findById ( id ).get ( );
-            orderDB.setBeverages ( order.getBeverages ( ) );
-            orderDB.setSize ( order.getSize ( ) );
-            orderDB.setSugar ( order.getSugar ( ) );
-            orderRepository.save ( orderDB );
-        }
-        return "Order " + id + " not exist";
-
+    public String updateOrder(UUID id, Order order) {
+        Order orderDB = orderRepository.findById(id).get();
```

Comment:
you shouldn't use get() here, you're receiving Optional you should do a check for if its found, check in Todo there is a construction with Lambda that you can use.

Reply:
In my opinion, I am doing many reduntant operations here. I got an order from bd, set the status, and save it to DB. I can simply add SQL query for update status in DB.